### PR TITLE
Refactoring NLP classifiers, making them all exportable, adding/movin…

### PR DESCRIPTION
**Pull request by borisfom**
…g unit tests

Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

### CODE REVIEW

File: nemo/collections/nlp/modules/common/classifier.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        return {"hidden_states": NeuralType(('B', 'T', 'D'), ChannelType())}
+
+    def __init__(self, hidden_size: int, dropout: float = 0.0,) -> None:
+
+        """
+        Initializes the Token Classifier module.
+
+        Args:
+            hidden_size: the size of the hidden dimension
+            num_classes: number of classes
```
Comment (okuchaiev) [inline nemo/collections/nlp/modules/common/classifier.py:46]: this docstring doesn't correspond to the actual args in \__init\__

File: nemo/collections/nlp/modules/common/sequence_classifier.py [hunk 2]
```diff
@@ -14,20 +14,15 @@
 
 from typing import Dict, Optional
 
-from torch import nn
-
-from nemo.collections.common.parts import MultiLayerPerceptron, transformer_weights_init
-from nemo.core.classes import NeuralModule, typecheck
-from nemo.core.neural_types import ChannelType, LogitsType, NeuralType
+from .classifier import Classifier
```
Comment (okuchaiev) [inline nemo/collections/nlp/modules/common/sequence_classifier.py:17]: please use full path: (e.g. nemo.bla.bla. )

File: nemo/collections/nlp/modules/common/sequence_regression.py [hunk 3]
```diff
@@ -14,16 +14,17 @@
 
 from typing import Dict, Optional
 
-from torch import Tensor, nn
+from torch import Tensor
 
-from nemo.collections.common.parts import MultiLayerPerceptron, transformer_weights_init
-from nemo.core.classes import NeuralModule, typecheck
-from nemo.core.neural_types import ChannelType, NeuralType, RegressionValuesType
+from .classifier import Classifier
```
Comment (okuchaiev) [inline nemo/collections/nlp/modules/common/sequence_regression.py:19]: please full path

File: nemo/collections/nlp/modules/common/sequence_token_classifier.py [hunk 4]
```diff
@@ -14,22 +14,17 @@
 
 from typing import Dict, Optional
 
-from torch import nn
-
-from nemo.collections.common.parts import MultiLayerPerceptron, transformer_weights_init
-from nemo.core.classes import NeuralModule, typecheck
-from nemo.core.neural_types import ChannelType, LogitsType, NeuralType
+from .classifier import Classifier
```
Comment (okuchaiev) [inline nemo/collections/nlp/modules/common/sequence_token_classifier.py:17]: please full path

File: nemo/collections/nlp/modules/common/token_classifier.py [hunk 5]
```diff
@@ -17,27 +17,21 @@
 import torch
 from torch import nn as nn
 
-from nemo.collections.common.parts import MultiLayerPerceptron, transformer_weights_init
-from nemo.core.classes import Exportable, NeuralModule, typecheck
-from nemo.core.neural_types import ChannelType, LogitsType, NeuralType
+from .classifier import Classifier
```
Comment (okuchaiev) [inline nemo/collections/nlp/modules/common/token_classifier.py:20]: please full path

File: nemo/collections/nlp/modules/common/sequence_regression.py (hunk 3 above)
Comment (ekmb) [reply nemo/collections/nlp/modules/common/sequence_regression.py:19]: full path?

File: nemo/collections/nlp/modules/common/classifier.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        Args:
+            hidden_size: the size of the hidden dimension
+            dropout: dropout to apply to the input hidden states
+        """
+        super().__init__()
+        self._hidden_size = hidden_size
+        self.dropout = nn.Dropout(dropout)
+
+    def post_init(self, use_transformer_init):
+        """
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/classifier.py:53]: please add type hint + docstring

File: nemo/collections/nlp/modules/common/common_utils.py [hunk 7]
```diff
@@ -61,8 +61,6 @@ def get_pretrained_lm_model(
             model, default_checkpoint_file = get_megatron_lm_model(
                 config_file=config_file, pretrained_model_name=pretrained_model_name
             )
-            if not checkpoint_file:
-                checkpoint_file = default_checkpoint_file
         else:
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/common_utils.py:64]: Why are you removing this? I was referring to https://github.com/NVIDIA/NeMo/blob/candidate/nemo/collections/nlp/modules/common/megatron/megatron_utils.py#L90 when was talking about an extra download call. Let's leave it for a separate PR though.

Comment (borisfom) [reply nemo/collections/nlp/modules/common/common_utils.py]: Sorry sneaked in here from another branch.
